workspaces/ui-v2/src/entry-points/local/local-cli.tsx

```diff
@@ -115,11 +115,17 @@ export function useLocalCliServices(
 ): AsyncStatus<LocalCliServices> {
   const opticEngine = useOpticEngine();
   const apiBaseUrl = `/api/specs/${specId}`;
-  const spectacle = useMemo(
-    () => new LocalCliSpectacle(apiBaseUrl, opticEngine),
-    [apiBaseUrl, opticEngine]
+  const [spectacle, setSpectacle] = useState(
+    new LocalCliSpectacle(apiBaseUrl, opticEngine)
   );
-  const capturesService = React.useMemo(
+  const refEngine = useRef(opticEngine);
+  const refApiUrl = useRef(apiBaseUrl);
+  useEffect(() => {
+    spectacle.registerSpectacleUpdateEvent(() => {
+      setSpectacle(new LocalCliSpectacle(refApiUrl.current, refEngine.current));
+    });
```

Comment:
We'll want to return [an effect clean up function here](https://reactjs.org/docs/hooks-effect.html#example-using-hooks-1), so we can tidy up the event listener. Might look like a nit pick, but bugs caused due to unregistered event emitters tend to show themselves in super obscure ways and can be a massive PITA.

Reply:
Bleh - I was like I need to handle this and totally forgot about useEffect's teardown yesterday (was frustrated with the internet being out!) Thanks for the reminder!